src/vulkan-renderer/application.cpp

```diff
@@ -333,7 +341,59 @@ Application::Application(int argc, char **argv) {
             auto debug_report_ci = wrapper::make_info<VkDebugReportCallbackCreateInfoEXT>();
             debug_report_ci.flags = VK_DEBUG_REPORT_ERROR_BIT_EXT | VK_DEBUG_REPORT_WARNING_BIT_EXT |
                                     VK_DEBUG_REPORT_PERFORMANCE_WARNING_BIT_EXT | VK_DEBUG_REPORT_ERROR_BIT_EXT;
-            debug_report_ci.pfnCallback = (PFN_vkDebugReportCallbackEXT)&vulkan_debug_message_callback;
+
+            // To avoid complex dependencies to other modules, we use a lambda here to report the validation layer
+            // message. We are using two different versions of the lambda because we can't pass
+            // stop_on_validation_message to the lambda directly.
+            if (stop_on_validation_message) {
+                const auto vulkan_validation_layer_callback = reinterpret_cast<PFN_vkDebugReportCallbackEXT>(
+                    +[](VkDebugReportFlagsEXT flags, VkDebugReportObjectTypeEXT object_type, std::uint64_t object,
+                        std::size_t location, std::int32_t message_code, const char *layer_prefix, const char *message,
+                        void *user_data) {
+                        if (flags & VK_DEBUG_REPORT_INFORMATION_BIT_EXT) {
+                            spdlog::info(message);
+                        } else if (flags & VK_DEBUG_REPORT_DEBUG_BIT_EXT) {
+                            spdlog::debug(message);
+                        } else if (flags & VK_DEBUG_REPORT_ERROR_BIT_EXT) {
+                            spdlog::error(message);
+                        } else if (flags & VK_DEBUG_REPORT_PERFORMANCE_WARNING_BIT_EXT) {
+                            spdlog::warn(message);
+                        } else {
+                            spdlog::warn(message);
+                        }
+
+                        // This feature stops command lines from overflooding with messages in case many validation
+                        // layer messages are reported in a short amount of time.
+                        spdlog::critical("Command line argument --stop-on-validation-message is enabled.");
+                        spdlog::critical("Application will cause a break point now!");
+
+                        // Wait for spdlog to shut down before aborting.
+                        spdlog::shutdown();
+                        std::abort();
+                        return VK_FALSE;
+                    });
+                debug_report_ci.pfnCallback = (PFN_vkDebugReportCallbackEXT)vulkan_validation_layer_callback;
```

Comment:
```suggestion
                debug_report_ci.pfnCallback = reinterpret_cast<PFN_vkDebugReportCallbackEXT>(
                    +[](VkDebugReportFlagsEXT flags, VkDebugReportObjectTypeEXT object_type, std::uint64_t object,
                        std::size_t location, std::int32_t message_code, const char *layer_prefix, const char *message,
                        void *user_data) {
                        if (flags & VK_DEBUG_REPORT_INFORMATION_BIT_EXT) {
                            spdlog::info(message);
                        } else if (flags & VK_DEBUG_REPORT_DEBUG_BIT_EXT) {
                            spdlog::debug(message);
                        } else if (flags & VK_DEBUG_REPORT_ERROR_BIT_EXT) {
                            spdlog::error(message);
                        } else if (flags & VK_DEBUG_REPORT_PERFORMANCE_WARNING_BIT_EXT) {
                            spdlog::warn(message);
                        } else {
                            spdlog::warn(message);
                        }

                        // This feature stops command lines from overflooding with messages in case many validation
                        // layer messages are reported in a short amount of time.
                        spdlog::critical("Command line argument --stop-on-validation-message is enabled.");
                        spdlog::critical("Application will cause a break point now!");

                        // Wait for spdlog to shut down before aborting.
                        spdlog::shutdown();
                        std::abort();
                        return VK_FALSE;
                    });
```

Could you try this?

Reply:
What's the difference? I can't see it

Reply:
I dont use the variable.

Reply:
I replaced both lambda variables with an inlined lambda.

Reply:
Done.